### DISCUSSION TITLE
Is it possible that using the jazzer fuzz the Android thiry party

### DISCUSSION THREAD

**Discussion by william31212** [CodeIntelligenceTesting/jazzer#413] [Q&A]
Hi, is it possible using the jazzer fuzz android third party. For instance, [glide](https://github.com/bumptech/glide) is android image parser third party. However, I am not sure that the bottom layer is different (dalvik vm, jvm) will make jazzer cannot fuzz it.

**Comment by fmeum** [comment]
I don't have a good answer to this at this point in time since I am not very familiar with the foundations of Java on Android. I'm planning to improve on that state in the upcoming weeks and will post an update if I have one.

**Reply by william31212** [reply]
Ok, Thank you for the quick reply.

**Reply by fmeum** [reply]
This is now being worked on by @TheCoryBarker.

**Reply by TheCoryBarker** [reply]
@william31212 is this something you are still interested in? Do you want me to keep you updated?

I am working on getting the running Jazzer on Android checked in. I have a PR out for this but I am also going to need to follow this up with another check in for offline instrumentation before it will be easy to fuzz Android.

**Reply by william31212** [reply]
@TheCoryBarker I am still interested in fuzzing Android. Is your PR mention on the Android's SDK or other target?

**Reply by TheCoryBarker** [reply]
Here is my PR:
https://github.com/CodeIntelligenceTesting/jazzer/pull/587

after this I need to update Jazzer to instrument offline. I'm starting the offline instrumentation next week, and I expect it will be fairly quick so I should have a PR out for it by next Friday. I'll send you an update once this is done.

**Reply by ghost** [reply]
Hi @william31212 and @TheCoryBarker ! 
I know this was over a year a go, but how did it turn out? Did the pull request get accepted? 
Happy to explore this in more detail with each of you. 
We can also take this "offline" over email. I am david.merian[at]code-intelligence[dot]com

Cheers,
Dave

**Reply by TheCoryBarker** [reply]
It's very difficult to use at this stage without a lot of knowledge on how APKs are build. I'm working on another PR that will make fuzzing your Android apps easy. I'll send you an email when it's ready David and then we can circle back and update here

**Reply by bay0max** [reply]
> It's very difficult to use at this stage without a lot of knowledge on how APKs are build. I'm working on another PR that will make fuzzing your Android apps easy. I'll send you an email when it's ready David and then we can circle back and update here

Hi, is there any progress,Did the pull request get accepted?

**Comment by TheCoryBarker** [comment]
It was coming along, but ultimately we stopped working on fuzzing Java
for Android. It was hard to see the value in completing it, since:
1: Static analysis (like CodeQL) is much easier for Java. It already works,
you don't have to worry about code coverage, and it comes with a lot of out
of the box queries to find vulnerabilities.
2: There are no Android specific sanitizers in Jazzer, so even after adding
it to the build system we still need to write all new sanitizers.

Since it was a project with questionable impact for Android (delta between
what exists already and what exists already + Java fuzzing), it ended up
getting stopped.
